tensorflow/compiler/xla/mlir_hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -3553,6 +3553,54 @@ LogicalResult ClampOp::reifyReturnTypeShapes(
                                 &reifiedReturnShapes);
 }
 
+
+OpFoldResult ClampOp::fold(ArrayRef<Attribute> operands) {
+  auto val = operands[0].dyn_cast_or_null<DenseElementsAttr>();
+  if (!val) return {};
+  DenseElementsAttr min_val = operands[1].dyn_cast_or_null<DenseElementsAttr>();
+  if (!min_val) return {};
+  DenseElementsAttr max_val = operands[2].dyn_cast_or_null<DenseElementsAttr>();
+  if (!max_val) return {};
+
+  auto type = getElementTypeOrSelf(getType());
+  if (!type.isF32() && !type.isF64()) return {};
+
+  auto shapedType = getType().cast<ShapedType>();
+  if (!shapedType.hasStaticShape()) return {};
+
+  // Prevent folding if the result is too large.
+  if (val.getNumElements() > kFoldOpEltLimit) return {};
+
+  // min/max/val should have the same shape
+  int64_t val_num = val.getNumElements();
+  int64_t min_num = min_val.getNumElements();
+  int64_t max_num = max_val.getNumElements();
+  if (val_num != min_num || val_num != max_num) return {};
```

Comment:
Support the scalar case as well?
![image](https://user-images.githubusercontent.com/15364516/198475578-3edc3762-065d-4fbb-843d-3bb0366c2bcc.png)